.ci/release/Jenkinsfile

```
@@ -0,0 +1,229 @@
+#!/usr/bin/env groovy
+
+@Library('apm@current') _
+pipeline {
+  agent { label 'linux && immutable' }
+  environment {
+    REPO = 'apm-agent-java'
+    BASE_DIR = "src/github.com/elastic/${env.REPO}"
+    NOTIFY_TO = credentials('notify-to')
+    DOCKERHUB_SECRET = 'secret/apm-team/ci/elastic-observability-dockerhub'
+    ELASTIC_DOCKER_SECRET = 'secret/apm-team/ci/docker-registry/prod'
+    NEXUS_SECRET = 'secret/apm-team/ci/nexus'
+    MAVEN_CONFIG = '-B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn -Dmaven.repo.local=.m2'
+    HOME = "${env.WORKSPACE}"
+    JAVA_HOME = "${env.HUDSON_HOME}/.java/java10"
+    PATH = "${env.JAVA_HOME}/bin:${env.PATH}"
+  }
+  options {
+    timeout(time: 3, unit: 'HOURS')
+    buildDiscarder(logRotator(numToKeepStr: '20', artifactNumToKeepStr: '20', daysToKeepStr: '30'))
+    timestamps()
+    ansiColor('xterm')
+    durabilityHint('PERFORMANCE_OPTIMIZED')
+
+  }
+  parameters {
+    string(name: 'branch_specifier', defaultValue: 'master')
+  }
+
+  stages {
+    stage('Initializing'){
+      options { skipDefaultCheckout() }
+      stages {
+        stage('Checkout') {
+          steps {
+            gitCheckout(
+                basedir: "${BASE_DIR}",
+                branch: 'master',
+                repo: 'git@github.com:elastic/apm-agent-java.git',
+                credentialsId: 'f6c7695a-671e-4f4f-a331-acdce44ff9ba',
+                shallow: false
+            )
+            stash(allowEmpty: true, name: 'source', useDefaultExcludes: false)
+            dir("${BASE_DIR}") {
+              setupAPMGitEmail()
+              script {
+                env.TAG_BARE = sh(script: "git tag | tail -1", returnStdout: true)
+                env.TAG_VER = sh(script: "git tag | tail -1 | sed s/v//", returnStdout: true)
+                env.TAG_DOT_X = sh(script: 'git tag|tail -1|sed s/v//|cut -f1 -d "."|awk \'{print $1".x"}\'', returnStdout: true)
+              }
+            }
```

What do you think to move the setupAPMGitEmail before the stash? Then all the other calls afteeward are not required:

```suggestion
            dir("${BASE_DIR}") {
              setupAPMGitEmail()
            }
            stash(allowEmpty: true, name: 'source', useDefaultExcludes: false)
            dir("${BASE_DIR}") {
              script {
                env.TAG_BARE = sh(script: "git tag | tail -1", returnStdout: true)
                env.TAG_VER = sh(script: "git tag | tail -1 | sed s/v//", returnStdout: true)
                env.TAG_DOT_X = sh(script: 'git tag|tail -1|sed s/v//|cut -f1 -d "."|awk \'{print $1".x"}\'', returnStdout: true)
              }
            }
```

---

I was wondering about that. I was just being extra-cautious here since there isn't much harm in setting it multiple times but if it will work this way than that's even better.